pilot/pkg/networking/plugin/authz/rbac_v2.go

```diff
@@ -77,6 +85,15 @@ func convertRbacRulesToFilterConfigV2(service *serviceMetadata, option rbacOptio
 			if binding.RoleRef != nil {
 				role = option.authzPolicies.RoleForNameAndNamespace(binding.RoleRef.Name, namespace)
 				serviceRoleName = binding.RoleRef.Name
+			} else if binding.Role != "" {
+				if strings.HasPrefix(binding.Role, serviceRoleRootNamespaceRef) {
+					localRoleName := binding.Role[1:]
```

Comment:
You can just `serviceRoleName = strings.TrimPrefix(binding.Role, serviceRoleRootNamespaceRef)`

Reply:
Good. Thanks!